docs/designs/login-design.md

```diff
@@ -0,0 +1,165 @@
+- Simple PIN-based authentication for volunteers.
+- Data integrity and security for volunteer information.
+- Mechanism for PIN recovery via email.
+
+
+### 3.2 Constraints
+- Must use Azure OAuth for authentication.
+- Admin and volunteer dashboards are outside the scope of this design.
+- The solution must comply with data protection and privacy regulations.
+- The process for resetting a volunteer's PIN has not been finalized. It could be managed by admins via the admin page or by the volunteers themselves through a new PIN reset page.
+
+## 4. Architectural Representations
+
+### 4.1 High-Level Architecture Diagram
+The high-level architecture includes the following components:
+- **Sign-In Page**: Redirects to Azure OAuth.
+- **Azure OAuth Service**: Authenticates users based on their email.
+- **Volunteer Name Page**: Allows volunteers to select name using a dropdown.
+- **Volunteer PIN Page**: Allows volunteers to sign in using a PIN.
+- **Email Notification**: Sends email notifications for user to reset the PIN.
+- **Database**: Stores volunteer details, including the PIN, activation status, and last sign-in date.
+
+### 4.2 Component Descriptions
+- **Sign-In Page**: Initial landing page for all users; redirects to Azure OAuth for authentication.
+- **Azure OAuth Service**: Handles secure authentication; determines user role based on the email domain.
+- **Volunteer Name Page**: Displays a dropdown of volunteers names.
+- **Volunteer PIN Page**: Displays a PIN field; checks entered data against the database.
+- **Email Notification via Resend API**:  Sends email notifications using the Resend API through Supabase Edge Functions. This feature is used for sending important notifications, such as PIN reset emails to volunteers. [Resend API Documentation](https://resend.com/docs/send-with-supabase-edge-functions)
+- **Database**: Manages volunteer data, including name, email address, PIN, and last sign-in date.
+
+## 5. Detailed Design
+
+### 5.1 Subsystems and Components
+#### Sign-In Page
+- **Purpose**: Redirect users to Azure OAuth for authentication.
+- **Input**: User clicks the Azure sign-in button.
+- **Output**: Redirect to Azure OAuth login page.
+- **Dependencies**: Azure OAuth Service.
+
+#### Volunteer Name Page
+- **Purpose**: Allows volunteers to choice their name using a dropdown selection.
+- **Input**: Selected volunteer name.
+- **Output**: Redirect to the volunteer Volunteer PIN Page.
+- **Dependencies**: Database for volunteer details.
+- **Additional Features**:
+  - **“I don’t see my name, what do I do now”** link: When clicked, a dialog will pop up. Provides a phone number or email of a PH admin that can sort it out (either get them activated in the database, and/or let them into another account for urgent matters)
+  - **Active User Filtering**: Only volunteers marked as "active" in the database will appear in the dropdown selection. This ensures that only currently active volunteers can sign in.
```

Comment:
Would this flag turn FALSE after 90 days?

Reply:
It sounds like a good idea.